pkg/operator/event.go

```diff
@@ -0,0 +1,68 @@
+/*
+Copyright 2016 The Rook Authors. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+Some of the code below came from https://github.com/coreos/etcd-operator
+which also has the apache 2.0 license.
+*/
+package operator
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+
+	"k8s.io/client-go/pkg/api/unversioned"
+	kwatch "k8s.io/client-go/pkg/watch"
+)
+
+type Event struct {
+	Type   kwatch.EventType
+	Object *Cluster
+}
+
+type rawEvent struct {
+	Type   kwatch.EventType
+	Object json.RawMessage
+}
+
+func pollEvent(decoder *json.Decoder) (*Event, *unversioned.Status, error) {
```

Comment:
I was referring to the polling util methods. Kuberenetes provides a bunch of utils files that are normally used by kubernetes dev use when developing services/apps on kubernetes. 

Here are some example of where they are used. https://github.com/kubernetes/kubernetes/blob/master/pkg/volume/rbd/rbd.go#L26
https://github.com/kubernetes/kubernetes/blob/master/cmd/kubelet/app/server.go#L43
https://github.com/kubernetes/kubernetes/blob/master/cmd/kube-controller-manager/app/controllermanager.go#L37